lib/resources/aide_conf.rb

```diff
@@ -0,0 +1,182 @@
+# encoding: utf-8
+# author: Jen Burns, burnsjennifere@gmail.com
+
+require 'utils/filter'
+require 'utils/parser'
+
+# rubocop:disable Metrics/ClassLength
+module Inspec::Resources
+  class AideConf < Inspec.resource(1)
+    name 'aide_conf'
+    desc 'Use the aide_conf InSpec audit resource to test the rules established for
+      the file integrity tool AIDE. Controlled by the aide.conf file typically at /etc/aide.conf.'
+    example "
+    describe aide_conf do
+      its('selection_lines') { should include '/sbin' }
+    end
+
+    describe aide_conf.where { selection_line == '/bin' } do
+      its('rules.flatten') { should include 'r' }
+    end
+
+    describe aide_conf.all_have_rule('sha512') do
+      it { should eq true }
+    end
+    "
+
+    attr_reader :params
+
+    include CommentParser
+
+    def initialize(aide_conf_path = nil)
+      return skip_resource 'The `aide_conf` resource is not supported on your OS.' unless inspec.os.linux?
+      @conf_path = aide_conf_path || '/etc/aide.conf'
+      @content = nil
+      @rules = nil
+      read_content
+    end
+
+    def all_have_rule(rule)
+      # Case when file didn't exist or perms didn't allow an open
+      if @content.instance_of?(String)
+        return false
+      end
+      in_all_lines = true
+      all_lines = parse_conf(@content)
+      all_lines.each do |line|
+        if !line['rules'].include? rule
+          in_all_lines = false
+          break
+        end
+      end
+      in_all_lines
+    end
+
+    filter = FilterTable.create
+    filter.add_accessor(:where)
+          .add_accessor(:entries)
+          .add(:selection_lines, field: 'selection_line')
+          .add(:rules,           field: 'rules')
+
+    filter.connect(self, :params)
+
+    private
+
+    def read_content
+      @content = ''
+      @rules = {}
+      @content = read_file(@conf_path)
+      # Return if @content consists of skip_resource message
+      if @content.instance_of?(String)
```

Comment:
I removed helper read_file function and just moved that code into this function to help clarify and clean up this code.